src/fillTank.test.js

```diff
@@ -1,11 +1,188 @@
 'use strict';
 
 describe('fillTank', () => {
-  // const { fillTank } = require('./fillTank');
+  const { fillTank } = require('./fillTank');
 
-  it('should ', () => {
+  it('should be declared', () => {
+    expect(fillTank).toBeInstanceOf(Function);
+  });
+
+  it('should fill tank fully with no amout', () => {
+    const customer = {
+      money: 3000,
+      vehicle: {
+        maxTankCapacity: 40,
+        fuelRemains: 8,
+      },
+    };
+
+    fillTank(customer, 50);
+
+    expect(customer.money).toBe(1400);
+    expect(customer.vehicle.fuelRemains).toBe(40);
+  });
+
+  it('should fill tank fully with amout bigger than tank capacity', () => {
+    const customer = {
+      money: 3000,
+      vehicle: {
+        maxTankCapacity: 100,
+        fuelRemains: 21,
+      },
+    };
+
+    fillTank(customer, 29, 110);
+
+    expect(customer.money).toBe(709);
+    expect(customer.vehicle.fuelRemains).toBe(100);
+  });
+
+  it('should correctly withdraw money and fill the tank', () => {
+    const customer = {
+      money: 2000,
+      vehicle: {
+        maxTankCapacity: 40,
+        fuelRemains: 12,
+      },
+    };
 
+    fillTank(customer, 30, 26);
+
+    expect(customer.money).toBe(1220);
+    expect(customer.vehicle.fuelRemains).toBe(38);
+  });
+
+  it('should fill only within the budget', () => {
+    const customer = {
+      money: 500,
+      vehicle: {
+        maxTankCapacity: 40,
+        fuelRemains: 20,
+      },
+    };
+
+    fillTank(customer, 100, 20);
+
+    expect(customer.money).toBe(0);
+    expect(customer.vehicle.fuelRemains).toBe(25);
   });
 
```

Comment:
The test description 'should round remaining money to hundreth part' contains a typo. It should be 'hundredth' instead of 'hundreth'.

Reply:
done